tests/optimagic/optimizers/test_bayesian_optimizer.py

```diff
@@ -0,0 +1,145 @@
+"""Unit tests for Bayesian optimizer helper functions."""
+
+import numpy as np
+import pytest
+
+from optimagic.config import IS_BAYESOPT_INSTALLED
+from optimagic.optimization.internal_optimization_problem import InternalBounds
+
+if IS_BAYESOPT_INSTALLED:
+    from bayes_opt import acquisition
+
+    from optimagic.optimizers.bayesian_optimizer import (
+        _extract_params_from_kwargs,
+        _process_acquisition_function,
+        _process_bounds,
+    )
+
+
+def test_extract_params_from_kwargs():
+    """Test extracting parameters from kwargs dictionary."""
+    kwargs = {"param0": 1.0, "param1": 2.0, "param2": 3.0}
+
+    result = _extract_params_from_kwargs(kwargs, 3)
+    np.testing.assert_array_equal(result, np.array([1.0, 2.0, 3.0]))
+
+    result = _extract_params_from_kwargs(kwargs, 2)
+    np.testing.assert_array_equal(result, np.array([1.0, 2.0]))
+
+
+def test_process_bounds():
+    """Test processing bounds for Bayesian optimization."""
+    bounds = InternalBounds(lower=np.array([-1.0, 0.0]), upper=np.array([1.0, 2.0]))
+    result = _process_bounds(bounds)
+    expected = {"param0": (-1.0, 1.0), "param1": (0.0, 2.0)}
+    assert result == expected
+
+    bounds_none = InternalBounds(lower=None, upper=np.array([1.0, 2.0]))
+    with pytest.raises(
+        ValueError, match="Bayesian optimization requires finite bounds"
+    ):
+        _process_bounds(bounds_none)
+
+    bounds_inf = InternalBounds(
+        lower=np.array([-1.0, 0.0]), upper=np.array([1.0, np.inf])
+    )
+    with pytest.raises(
+        ValueError, match="Bayesian optimization requires finite bounds"
+    ):
+        _process_bounds(bounds_inf)
```

Comment:
Let's split this into three tests for the three cases so we get more information if any of them fails.